### PR TITLE
Fix SortableList resting row layout

### LINKED ISSUE #932: SortableList: items have no row layout, so drag handles stack below their labels
## Symptom
In SortableList, each item's drag handle appears stacked below its label instead of positioned alongside it in a row.

## Root cause
`sortable-list.css:13-17` is the complete base rule for the item container:
```
.cinder-sortable-item {
  position: relative;
  cursor: default;
  user-select: none;
}
```
No `display: flex`/`grid`, no `align-items`, no `gap`. The item is a plain block container, so its children (label, handle) stack vertically per normal block flow.

The handle itself is correctly styled in isolation — `.cinder-sortable-handle { display: inline-flex; align-items: center; }` (`sortable-list.css:71-73`) — but nothing on the parent positions it relative to the label.

By contrast, the drag *interaction* states are thoroughly styled: `--lifted` (`:19`), `--placeholder` (`:38`), `--shifting` (`:52`), with reduced-motion variants (`:57-66`). The interaction model got attention; the resting layout did not.

## Fix
Make `.cinder-sortable-item` a row container (flex or grid), with `align-items: center` and a defined `gap`. Place the handle consistently with how `selectable-row` and `stacked-list-item` position their leading/trailing affordances, so SortableList doesn't invent a third convention.

After the layout fix, re-verify the lifted/placeholder/shifting states, since they were authored against the current (broken) box model and may need adjustment once the item becomes a flex/grid row.

## Acceptance criteria
- Handle and label sit in a single row with consistent alignment and spacing, matching the leading/trailing affordance convention used elsewhere.
- Lifted/placeholder/shifting states still render correctly against the corrected layout.

## Regression test
Add a resting-state visual regression test for SortableList — none currently exists for static (non-dragging) layout. Assert handle and label share a row (e.g., comparable `offsetTop`/bounding-rect vertical center) rather than the handle sitting below the label.

### PR DESCRIPTION
**Pull request by stevekinney**
Closes #932

## Summary

- group arbitrary item snippet content into a shrink-safe content cell
- lay out each SortableList item as a two-column row with a centered trailing drag handle
- add a real-browser geometry regression that proves label and handle centers share a row

## Validation

- `bun test packages/components/src/components/sortable-list/sortable-list.test.ts`
- `bun run --filter=@cinder/testing test:playwright -- tests/sortable-list-layout.playwright.ts`
- `bun run --filter=@lostgradient/cinder typecheck`
- `bun run --filter=@cinder/testing typecheck`
- `bunx stylelint packages/components/src/components/sortable-list/sortable-list.css`
- `bunx prettier --check packages/components/src/components/_sortable-item.svelte packages/components/src/components/sortable-list/sortable-list.css packages/testing/tests/sortable-list-layout.playwright.ts`

### CODE REVIEW

**Comment by stevekinney** [comment]
Convergence summary

- Fixed the resting SortableList row layout and added a real-browser geometry regression.
- Addressed both Codex review threads by moving wrapper/layout ownership out of the shared `_sortable-item` primitive and into SortableList-specific classes.
- Added a Kanban browser regression proving the all-styles cascade preserves `align-items: start` and does not inject the SortableList content wrapper.
- Resolved both review threads; no follow-up issues were deferred.
- Exact head `b0040940cf15232e2d2eb5024116398fc4b21bfc`: unit tests, typecheck, Playwright, changeset guard, and Vercel are terminal and green. Two fresh current-main readings report `MERGEABLE` / `CLEAN`, zero unresolved threads, and a clean merge tree.

File: packages/components/src/components/sortable-list/sortable-list.css [hunk 1]
```diff
@@ -10,6 +10,17 @@
     padding: 0;
   }
 
+  .cinder-sortable-list__item {
+    display: grid;
+    grid-template-columns: minmax(0, 1fr) auto;
+    align-items: center;
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/sortable-list/sortable-list.css:16]: Scoped the Kanban alignment rule to direct card handles (`.cinder-kanban-board__card > .cinder-sortable-handle`), so nested SortableList handles retain their own alignment. Focused Kanban tests pass (51/51).

File: packages/components/src/components/sortable-list/sortable-list.css [hunk 2]
```diff
@@ -10,11 +10,18 @@
...
 
-  [data-cinder-sortable-list] > .cinder-sortable-item,
-  [data-cinder-sortable-list-preview] > .cinder-sortable-item {
-    display: flex;
+  .cinder-sortable-list__item {
+    display: grid;
+    grid-template-columns: minmax(0, 1fr) auto;
     align-items: center;
-    gap: var(--cinder-space-2);
+    column-gap: var(--cinder-space-2);
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/sortable-list/sortable-list.css:17]: Fixed in 81c0f2e53: stale selector assertions now validate the retained sortable-list item/content grid classes and preview styling.

File: packages/components/src/components/sortable-list/sortable-list.css [hunk 3]
```diff
@@ -10,11 +10,18 @@
...
+  }
+
+  .cinder-sortable-list__item-content {
+    min-inline-size: 0;
+  }
+
+  .cinder-sortable-item {
     position: relative;
     cursor: default;
     user-select: none;
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/sortable-list/sortable-list.css]: Fixed in 81c0f2e53: cursor and user-select rules are scoped to cinder-sortable-list__item, preventing Kanban's shared cinder-sortable-item class from inheriting SortableList behavior.
